sopel/modules/reddit.py

```diff
@@ -328,3 +378,37 @@ def get_channel_spoiler_free(bot, trigger):
         bot.say('%s is flagged as spoiler-free' % channel)
     else:
         bot.say('%s is flagged as spoilers-allowed' % channel)
+
+
+@rule(r'^(?P<prefix>r|u)/(?P<id>[a-zA-Z0-9-_]+)')
```

Comment:
I played with some stuff and came up with (what I think is) an even better pattern.

```suggestion
@rule(r'.*\b(?P<prefix>r|u)/(?P<id>[a-zA-Z0-9-_]+)\b.*')
```

This will match a user or subreddit reference anywhere in the line, instead of only at the beginning, and properly handles things like `r/sopel's moderators` or `u/deathbybandaid'll get ya!`.

It only handles the _last_ reference in the line, though. That's a known issue with the `rule` decorator, which we'll probably work on for 7.1 (see #1757).

Reply:
Your proposed regex is giving me undesired double results, and not even from the "u/" and "r/", but full URL matching.

![image](https://user-images.githubusercontent.com/18010545/69350227-2ae3b980-0c47-11ea-9831-3c29cdb1512d.png)

Reply:
I would, though like suggestions for regex that will help me accomplish my second comment on this PR (for my next PR), which would allow users to type "r/plex hot" and get the first item in the sorting "hot"

Reply:
Discussed more in https://github.com/sopel-irc/sopel/pull/1722#issuecomment-557316789 but here's an improved pattern I came up with for this:

    (?i)(?<!\S)\/?(?P<prefix>r|u)\/(?P<id>[a-zA-Z0-9-_]+)\b

Reply:
In Sopel itself, we don't need the flag (`(?i)`), but we _do_ need to wrap the whole pattern in `.*` for it to work.

    @rule(r'.*(?<!\S)/?(?P<prefix>r|u)/(?P<id>[a-zA-Z0-9-_]+)\b.*')